src/server/editorServices.ts

```diff
@@ -345,7 +350,7 @@ namespace ts.server {
         /**
          * Container of all known scripts
          */
-        private readonly filenameToScriptInfo = createMap<ScriptInfo>();
+        private readonly filenameToScriptInfo = createMap<ScriptInfoOrVersion>();

```

Comment:
would it be simpler to have a different map of deleted scripts with their versions instead of a union type? we only need to touch that on add and delete.